WebService/v1/Controllers/ReplayFileController.cs

```diff
@@ -0,0 +1,114 @@
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Exceptions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Filters;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models.Helpers;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models.ReplayFileApiModel;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Controllers
+{
+    [ExceptionsFilter]
+    public class ReplayFileController : Controller
+    {
+        private const string TEXT_CSV = "text/csv";
+        private const string TYPE_CSV = "csv";
+        private readonly ILogger log;
+        private readonly IReplayFileService replayFileService;
+
+        public ReplayFileController(
+            IReplayFileService replayFileService,
+            ILogger logger)
+        {
+            this.replayFileService = replayFileService;
+            this.log = logger;
+        }
+
+        [HttpGet(Version.PATH + "/[controller]/{id}")]
+        public async Task<ReplayFileApiModel> GetAsync(string id)
+        {
+            return ReplayFileApiModel.FromServiceModel(await this.replayFileService.GetAsync(id));
+        }
+
+        [HttpPost(Version.PATH + "/[controller]!validate")]
+        public ActionResult Validate(IFormFile file)
+        {
+            if (file == null)
+            {
+                this.log.Warn("No data provided");
+                throw new BadRequestException("No data provided.");
+            }
+
+            if (file.ContentType != TEXT_CSV && !file.FileName.EndsWith(".csv"))
+            {
+                this.log.Warn("Wrong content type provided", () => new { file.ContentType });
+                throw new BadRequestException("Wrong content type provided.");
+            }
+
+            try
+            {
+                var content = this.replayFileService.Validate(file.OpenReadStream());
+            }
+            catch (Exception e)
+            {
+                return new JsonResult(new ValidationApiModel
+                    {
+                        IsValid = false,
+                        Messages = new List<string>
+                        {
+                            e.Message
+                        }
+                    })
+                    { StatusCode = (int) HttpStatusCode.BadRequest };
+            }
+
+            return new JsonResult(new ValidationApiModel()) { StatusCode = (int) HttpStatusCode.OK };
+        }
+
+        [HttpPost(Version.PATH + "/[controller]")]
+        public async Task<ReplayFileApiModel> PostAsync(IFormFile file)
+        {
+            if (file == null)
+            {
+                this.log.Warn("No data provided");
+                throw new BadRequestException("No data provided.");
+            }
+
+            if (file.ContentType != TEXT_CSV && !file.FileName.EndsWith(".csv"))
+            {
+                this.log.Warn("Wrong content type provided", () => new { file.ContentType });
```

Comment:
Can we return the expected content type in the error message? If I was calling this API and was told that I was using the wrong content type, I'd appreciate a message telling me what the expected content type is.

Reply:
sure